### PR TITLE
Allow reply from individual notifications on Android

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Refactored Android Push notifications. With this refactor we ensure that each notification is not merged with previous ones as we used to do, this allows a user to reply to a specific message in case there are multiple notifications from the same channel but different threads.

Following the [android documentation for push notification best practices](https://developer.android.com/training/notify-user/build-notification#messaging-best-practices):
> After a user replies with the inline reply action, use MessagingStyle.addMessage() to update the MessagingStyle notification and do not retract or cancel the notification. Not cancelling the notification allows a user to send multiple replies from the notification.

This PR also includes the sender avatar in the push notification and fixes an issue when trying to reply to a message that belongs to a thread.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-22933
https://mattermost.atlassian.net/browse/MM-36693

#### Screenshots
image

image

image

#### Release Note
```release-note
Android:
* Added ability to reply to individual messages using the push notifications
* Included sender avatar in push notifications
* Fixed an issue that cause the reply from a push notification to fail when the message belongs to a thread
```

### CODE REVIEW

**Comment by matthewbirtch** [comment]
@enahum is there a specific server that I need to log in with to test this properly?

**Comment by enahum** [comment]
@matthewbirtch no, community would do

**Review by matthewbirtch — APPROVED** [review]
Tested on my Android and the new functionality seems to work great! Verified that I get the proper avatar in the notification now and replying from the notification works as expected (with CRT on or off). Nice work @enahum

**Comment by esethna** [comment]
@enahum I haven't tested but this sounds like we're removing a pretty key piece of functionality that is stacking push notifications?

cc// @matthewbirtch

**Comment by enahum** [comment]
@esethna do watch the screenshots, they are grouped together, Also it would be good if you give the Android docs a read https://developer.android.com/training/notify-user/group

**Comment by enahum** [comment]
Also this work will help you handle CRT notifications with just a little tweak

**Comment by esethna** [comment]
@enahum Thanks, I'm just trying to understand how we used to group before vs how we are grouping now.

Based on the screenshots it looks like we are grouping by thread now? But how did we used to group before this change, by channel?

**Comment by enahum** [comment]
@esethna we are grouping by channel, not thread... what we use to do was replace the notification message with all the conversations that sent a notification for a specific channel

**Review by esethna — APPROVED** [review]
@enahum ah got it, thanks for helping me understand the change

File: package.json [hunk 1]
```diff
@@ -150,6 +151,7 @@
     "react-native-dev-menu": "4.0.2",
     "react-native-dotenv": "2.5.5",
     "react-native-storybook-loader": "2.0.4",
+    "react-test-renderer": "17.0.2",
```
Comment (saturninoabril) [inline package.json:154]: I don't see these new packages being used elsewhere.

**Review by saturninoabril — APPROVED** [review]
Tested and looks good to me.

File: package.json (hunk 1 above)
Comment (enahum) [reply package.json:154]: @saturninoabril without them tests are failing

**Review by anurag6713 — APPROVED** [review]
Thanks @enahum 🙂

**Pull request merged**